packages/client/src/usePhaserLayer.ts

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { NetworkLayer } from "./layers/network";
+import { createPhaserLayer } from "./layers/phaser";
+import { phaserConfig } from "./layers/phaser/config";
+import useResizeObserver, { ResizeHandler } from "use-resize-observer";
+import { throttle } from "lodash";
+import { usePromiseValue } from "./usePromiseValue";
+import { debug as parentDebug } from "./debug";
+
+const debug = parentDebug.extend("usePhaserLayer");
+
+const createContainer = () => {
+  const container = document.createElement("div");
+  container.style.width = "100%";
+  container.style.height = "100%";
+  container.style.pointerEvents = "all";
+  return container;
+};
+
+type Props = {
+  networkLayer: NetworkLayer | null;
+  containerId?: string;
+  hidden?: boolean;
+};
+
+export const usePhaserLayer = ({ networkLayer, hidden = false }: Props) => {
+  const parentRef = useRef<HTMLElement | null>(null);
+  const [{ width, height }, setSize] = useState({ width: 0, height: 0 });
+
+  const { phaserLayerPromise, container } = useMemo(() => {
+    if (!networkLayer) return { phaserLayerPromise: null, container: null };
+    debug("got new network layer");
+
+    debug("creating phaser layer");
+    const container = createContainer();
+    if (parentRef.current) {
+      parentRef.current.appendChild(container);
+    }
+    return {
+      container,
+      phaserLayerPromise: createPhaserLayer(networkLayer, {
+        ...phaserConfig,
+        scale: {
+          ...phaserConfig.scale,
+          parent: container,
+          // Phaser's default resize handling isn't great, so we'll do our own
+          // TODO: make a Phaser PR for this?
+          mode: Phaser.Scale.NONE,
```

Comment:
why do we set it to `RESIZE` in the config and then override it here?

Reply:
`RESIZE` seemed like a reasonable default, but since we're gonna replace the config value anyway, I'll get rid of that one.

I think leaving the `NONE` scale mode here makes the most sense, to tightly couple it to the resize handling behavior in the hook.